common/web/gesture-recognizer/src/zoneBoundaryChecker.ts

```diff
@@ -0,0 +1,67 @@
+namespace com.keyman.osk {
+  export class ZoneBoundaryChecker {
+    // This class exists for static methods & fields.
+    private constructor() { }
+
+    public static readonly FAR_TOP   : 0x0008 = 0x0008;
+    public static readonly FAR_LEFT  : 0x0004 = 0x0004;
+    public static readonly FAR_BOTTOM: 0x0002 = 0x0002;
+    public static readonly FAR_RIGHT : 0x0001 = 0x0001;
+
+    /**
+     * Determines the relationship of an input coordinate to one of the gesture engine's
+     * active recognition zones and returns a bitmask indicating which boundary (or
+     * boundaries) the input coordinate lies outside of.
+     *
+     * @param coord         An input coordinate
+     * @param zone          An object defining a 'recognition zone' of the gesture engine.
+     * @param ignoreBitmask A bitmask indicating select boundaries to ignore for the check.
+     */
+    static getCoordZoneBitmask(coord: InputEventCoordinate, zone: RecognitionZoneSource, ignoreBitmask?: number): number {
+      if(!ignoreBitmask) {
+        ignoreBitmask = 0;
+      }
```

Comment:
Given `ignoreBitmask` is only used in `inputMoveCancellationCheck()`, perhaps move the logic for it there and simplify the purpose of this function?